src/general/HexLayout.cs

```diff
@@ -220,7 +220,7 @@ public List<Hex> GetIslandHexes()
         var hexesWithNeighbours = new List<Hex> { initHex };
 
         // These are all of the existing hexes, that if there are no islands will all be visited
-        var shouldBeVisited = existingHexes.Select(p => p.Position).ToList();
+        var shouldBeVisited = ComputeHexCache();
```

Comment:
I think I see. The bug here was that as there are many organelles that are multiple hexes large, this didn't properly loop *all* hexes instead only the first hex of an organelle. Did I understand this right?

Reply:
Yes, exactly.